pypeit/specobjs.py

```diff
@@ -804,6 +804,10 @@ def write_info(self, outfile, pypeline):
                 opt_fwhm.append(np.median(specobj.FWHMFIT) * binspatial * platescale)
             else:  # Optimal is not required to occur
                 opt_fwhm.append(0.)
+            # TODO: S2N was None in the test Keck/KCWI reductions, which was
+            # causing the code to fault.  I (KBW) added this to get around it.
+            if specobj.S2N is None:
+                specobj.S2N = specobj.med_s2n()
```

Comment:
`specobjs` should only be used for the alignment frames for KCWI (but this doesn't seem to be the issue here). For KCWI, objects are found, but the extraction is not performed (this is because the DAR correction only gets applied during the construction of the datacube).

I'm not sure if this is the right solution, though... Probably (??) we don't want to save the information about the specobjs if we don't perform the extraction? These lines would probably need to be removed and a relevant catch added in `pypeit.py`... Something like the following on line 1022-ish:

`        if all_specobjs.nobj > 0 and not self.par['reduce']['extraction']['skip_extraction']:`

Reply:
Nice!  I've made your suggested change and running a test now.